testscript/testscript.go

```diff
@@ -69,6 +69,11 @@ func (e *Env) Defer(f func()) {
 	e.ts.Defer(f)
 }
 
+// T returns the current subtest.
```

Comment:
Interesting, all of those would work. For my use case `env.T().(testing.TB)` would be good enough, hence this proposal.

If instead we want to allow access to the subtest's `*testing.T`, that's different, as you described. But then we exclude `testscript.RunT`, because if you had a `*testing.T` in the first place, then you would have used `Run`. `RunT` is (I suppose) intended to be used when you don't actually have a `*testing.T`, but your own type implementing `testscript.T`.
If that's true, then we could implement something that's very easy to use for the relevant use case, and impossible to use otherwise. Like
```
// T returns the current subtest, or an error if testcript.RunT was used to run the tests.
// If Cleanup is called on the returned value, 
// the function will run after any functions passed to Env.Defer.
func (e *Env) T() (*testing.T, error)
```

Reply:
> For my use case env.T().(testing.TB) would be good enough, hence this proposal.

That's a reasonable point. Maybe there's not really a good use case for
making the actual concrete `*testing.T` value available.

Something like this, perhaps? (the same as your proposal
with an expanded doc comment)

```
// T returns the t argument passed to the current test by the T.Run method.
// Note that if the tests were started by calling Run,
// the returned value will implement testing.TB. Note that
// despite that, the underlying value will not be of type *testing.T
// because *testing.T does not implement T.
//
// If Cleanup is called on the returned value, the function will run
// after any functions passed to Env.Defer.
func (e *Env) T() T
```

If at some point, someone does come up with a specific use case that requires
`*testing.T`, we can always expose `TShim` or similar.

Reply:
Cool, I made that change to the docstring.